frontend/src/components/PropertyCard.vue

```diff
@@ -431,12 +439,117 @@ export default {
             }
           `,
         },
+      });
+
+      this.property = result.data.data.saleByCoordinatesId;
+    },
+
+    /**
+     * Fetches the property's estimated price
+     * @param {string} id - the property's ID
+     */
+    fetchEstimatedPrice(id) {
+      this.loading = true;
+      axios({
+        url: BACKEND_URL,
+        method: "post",
+        data: {
+          query: `
+            {
+              estimatedSalePriceById(id: "${id}")
+            }
+          `,
+        },
       }).then((result) => {
-        this.property = result.data.data.saleByCoordinatesId;
+        this.estimated_price = result.data.data.estimatedSalePriceById;
       }).finally(() => {
         this.loading = false;
       });
     },
+
+    /**
+     * Fetches the property's estimated price
+     * @param {string} id - the property's ID
+     */
+    fetchBreakEven(id) {
+      this.loading = true;
+      // At 100%
+      axios({
+        url: BACKEND_URL,
+        method: "post",
+        data: {
+          query: `
+            {
+              calculateBreakEven(id: "${id}", occupancyRate: 1.0)
+            }
+          `,
+        },
+      }).then((result) => {
+        this.break_even_100 = result.data.data.calculateBreakEven;
+        console.log("BE100:", this.break_even_100, result.data.data);
+      });
+
+      // At 80%
+      axios({
+        url: BACKEND_URL,
+        method: "post",
+        data: {
+          query: `
+            {
+              calculateBreakEven(id: "${id}", occupancyRate: 0.8)
+            }
+          `,
+        },
+      }).then((result) => {
+        this.break_even_80 = result.data.data.calculateBreakEven;
+        console.log("BE80:", this.break_even_80);
+      });
+
+      // At 60%
+      axios({
+        url: BACKEND_URL,
+        method: "post",
+        data: {
+          query: `
+            {
+              calculateBreakEven(id: "${id}", occupancyRate: 0.6)
+            }
+          `,
+        },
+      }).then((result) => {
+        this.break_even_60 = result.data.data.calculateBreakEven;
+        console.log("BE60:", this.break_even_60);
+      });
+    },
+
+    /**
+     * Fetches the property's computed rating
+     * @param {string} id - the property's ID
+     */
+    fetchRating(id) {
+      axios({
+        url: BACKEND_URL,
+        method: "post",
+        data: {
+          query: `
+            {
+              saleById(id: "${id}"){
+                score
+              }
+            }
+          `,
+        },
+      }).then((result) => {
+        console.log("Rating for", id);
+        // Get rating and apply to 1-10 range
+        if (result.data.data.saleById.score > 0) {
+          this.rating = Math.max(1, Math.round(result.data.data.saleById.score));
+        } else {
+          // Invalid rating
+          this.rating = 11;
+        }
+      });
+    },
```

Comment:
fetch with initial property fetching. Just add score to the list of fields

Reply:
The idea was to fetch this afterwards, since score calculation is kind of slow - or does it take the same time regardless of whether score is fetched?